Simplenote/NoteReferenceTableHeaderViewCell.swift

```diff
@@ -0,0 +1,60 @@
+import UIKit
+
+
+// MARK: - NoteReferenceTableHeaderViewCell
+//
+final class NoteReferenceTableHeaderViewCell: UITableViewCell {
```

Comment:
Nipticky: Would it be worth it renaming `NoteReferenceTableHeaderViewCell` > `DefaultTableViewCell` ?

That way we could have `Value1TableViewCell` / `SwitchTableViewCell` etc etc (maybe?)

Reply:
It would be awesome to have standardised cells. 
Renamed:
* `NoteReferenceTableHeaderViewCell` => `TableHeaderViewCell`
* `NoteReferenceTableViewCell` => `SubtitleTableViewCell`

Let's see if one day we can reuse them!